cypress/integration/v2/features/copy-paste-style.spec.js

```diff
@@ -0,0 +1,353 @@
+	'testimonial2',
+	'testimonial3',
+	// blog-posts
+	'numberOfItems',
+	'order',
+	'orderBy',
+	'postType',
+	'taxonomyType',
+	'taxonomy',
+	'taxonomyFilterType',
+	'postOffset',
+	'postExclude',
+	'postInclude',
+	// button, card, grid
+	'button1Url',
+	'button1NewTab',
+	'button1NoFollow',
+	'button2Url',
+	'button2NewTab',
+	'button2NoFollow',
+	'button3Url',
+	'button3NewTab',
+	'button3NoFollow',
+	'button4Url',
+	'button4NewTab',
+	'button4NoFollow',
+	// cta, notif
+	'buttonUrl',
+	'buttonNewTab',
+	'buttonNoFollow',
+	// card, feature, grid
+	'imageAlt',
+	'image1Alt',
+	'image2Alt',
+	'image3Alt',
+	'imageId',
+	'image1Id',
+	'image2Id',
+	'image3Id',
+	'image4Id',
+	'image1Url',
+	'image2Url',
+	'image3Url',
+	'image4Url',
+	// image box
+	'link1Url',
+	'link2Url',
+	'link3Url',
+	'link4Url',
+	'link1NewTab',
+	'link2NewTab',
+	'link3NewTab',
+	'link4NewTab',
+	'link1NoFollow',
+	'link2NoFollow',
+	'link3NoFollow',
+	'link4NoFollow',
+	// video popup
+	'previewBackgroundMediaUrl',
+	'previewBackgroundMediaId',
+	'videoLink',
+	'videoID',
+	'video-popup3',
+	// header
+	'showButton2',
+]
+
+function allBlocks() {
+	it( 'should copy & paste the attributes of the blocks', () => {
+		cy.setupWP()
+		cy.registerPosts( { numOfPosts: 1 } )
+		cy.newPage()
+
+		blocks
+			.forEach( blockName => {
+				const name = blockName.split( '/' ).pop()
+
+				cy.addBlock( blockName )
+				cy.fixture( `stackable/${ name }` ).then( block => {
+					cy.setBlockAttribute( block.attributes )
+					cy.getBlockAttributes().then( attributes1 => {
+						cy.addBlock( blockName )
+						// Enter some text into these blocks first
+						if ( name === 'text' ) {
+							cy.setBlockAttribute( {
+								'showTitle': true,
+								'showSubtitle': true,
+								'title': 'Text copy',
+								'subtitle': 'Text subtitle copy',
+								'text1': 'Lorem ipsum dolor',
+							} )
+						}
+						if ( name === 'heading' ) {
+							cy.typeBlock( blockName, `.ugb-${ name }__subtitle`, 'Subtitle', 1 )
+						}
+						if ( name === 'expand' ) {
+							cy.typeBlock( blockName, `.ugb-${ name }__title`, 'Title', 1 )
+						}
+						const blockAttributes = Object.keys( block.attributes )
+						const imageAttributes = [
+							'imageUrl',
+							'image1Url',
+							'image2Url',
+							'image3Url',
+							'image4Url',
+							'imageId',
+							'image1Id',
+							'image2Id',
+							'image3Id',
+							'image4Id',
+						]
+						const imageAttrsPresent = blockAttributes.filter( val => imageAttributes.includes( val ) )
+						// Set a different image URL for the 2nd block created for content assertion. Also set imageIds.
+						if ( imageAttrsPresent ) {
+							imageAttrsPresent.forEach( val => {
+								if ( val.includes( 'Id' ) ) {
+									cy.setBlockAttribute( {
+										[ val ]: 1010,
+									} )
+								} else {
+									cy.setBlockAttribute( {
+										[ val ]: Cypress.env( 'DUMMY_IMAGE_URL2' ),
+									} )
+								}
+							} )
+						}
```

Comment:
This is a very expensive call.

Avoid O(n^2) as much as possible. We can make this simpler.
You don't need to create a new array to store all attributes of the block.
```
imageAttributes.forEach(  imageAttribute => {
   if ( block.attributes.hasOwnProperty( imageAttribute ) ) {
       if ( imageAttribute.match( /Id/ ) {
           // set id attribute
           return
       }
       
       // set the URL attribute
   }
} )
```